jax/config.py

```diff
@@ -40,6 +42,15 @@ def int_env(varname: str, default: int) -> int:
   return int(os.getenv(varname, default))
 
 
+class _ThreadLocalState(threading.local):
```

Comment:
I'm wondering if it would be preferable to merge this state into the same ThreadLocalState object in api.py, where we store the jit is disabled flag.

Reply:
Originally I had it there, but it caused circular dependencies.

Reply:
I added a TODO to move the thread local state from api to config

Reply:
Another option would be somewhere like `core`.